sound/soc/sof/intel/hda.c

```diff
@@ -526,6 +545,8 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 
 	free_irq(sdev->ipc_irq, sdev);
 	free_irq(sdev->pci->irq, sdev);
+	if (sdev->msi_enabled)
```

Comment:
A quick check of the PCI code shows that the PCI APIs do the msi_enabled check internally so this is check not needed.